### PR TITLE
take data-root out of KateCommitment

### PR DESCRIPTION
**Pull request by fmiguelgarcia**

### CODE REVIEW

File: primitives/avail/src/header/extension/v2.rs [hunk 1]
```diff
@@ -13,11 +13,12 @@ use crate::{asdr::DataLookup, v2::KateCommitment};
 pub struct HeaderExtension {
 	pub commitment: KateCommitment,
 	pub app_lookup: DataLookup,
+	pub data_root: Option<H256>
```
Comment (fmiguelgarcia) [inline primitives/avail/src/header/extension/v2.rs:16]: We need to remove the `Option` to facilitate the reading of fixed size from the end.
`app_lookup` should be before `commitment` too, right?

File: primitives/avail/src/kate_commitment.rs [hunk 2]
```diff
@@ -76,21 +72,20 @@ pub mod v2 {
...
 				+ self.cols.size_of(ops)
-				+ self.data_root.size_of(ops)
 		}
 	}
 
-	#[cfg(test)]
-	mod tests {
-		use super::*;
-		use test_case::test_case;
+	// #[cfg(test)]
```
Comment (fmiguelgarcia) [inline primitives/avail/src/kate_commitment.rs:78]: Could we move this test to `extension/mod.rs`?

**Comment by markopoloparadox** [comment]
@fmiguelgarcia Hey,

Just a quick check-in on this PR. Is it still in progress?

**Comment by kroos47** [comment]
Misclicked this to open PR, can be closed instead

**Pull request closed without merging**